pom.xml

```diff
@@ -750,6 +752,35 @@
                 <classQualifiedName>com.fasterxml.jackson.databind.*</classQualifiedName>
                 <justification>Not relevant</justification>
               </item>
+              <item>
+                <code>java.annotation.removed</code>
+                <classQualifiedName>io.jenkins.plugins.analysis.core.steps.IssuesRecorder.Descriptor</classQualifiedName>
+                <justification>Symbol-based usage migrated transparently to RecordIssuesStep</justification>
+              </item>
+              <item>
+                <code>java.class.externalClassExposedInAPI</code>
+                <regex>true</regex>
+                <classQualifiedName> org.jenkinsci.plugins.workflow.*</classQualifiedName>
+                <justification>Classes must be exposed to implement a Pipeline Step</justification>
+              </item>
+              <item>
+                <code>java.*</code>
+                <regex>true</regex>
+                <classQualifiedName>org.kohsuke.stapler.*</classQualifiedName>
+                <justification>Classes from Stapler are not part of this plugin's API</justification>
+              </item>
+              <item>
+                <code>java.*</code>
+                <regex>true</regex>
+                <classQualifiedName>hudson.*</classQualifiedName>
+                <justification>Changes in Jenkins core are not part of this plugin's API</justification>
+              </item>
+              <item>
+                <code>java.*</code>
+                <regex>true</regex>
+                <classQualifiedName>jenkins.*</classQualifiedName>
+                <justification>Changes in Jenkins core are not part of this plugin's API</justification>
```

Comment:
This and the `hudson.*` exclusions are rather broad and at least `hudson.*` probably picks up some classes from detached plugins that are no longer in core, but otherwise it looked like the 2.121.x to 2.138.x core update would've involved maybe 50 individual ignores. I guess some of the other exclusions should be cleaned up now that they are covered by these catch-alls.

It doesn't really make sense to me to consider changes in Jenkins core (or any other provided-scope dependency) as part of the API of a plugin from the point of view of semantic versioning, since any plugin that depends on this plugin will itself have to depend on the same version of Jenkins core or newer, and the plugin is not providing Jenkins' classes, Jenkins is, so it would be Jenkins that would need to be semantically versioned.

Ideally we'd be able to add ignores on a Maven artifact level rather than a class level but it didn't look like that was possible.

It almost seems like it would make more sense to set [checkDependencies](https://revapi.org/modules/revapi-maven-plugin/check-mojo.html?#checkDependencies) to false and only consider changes to this plugin itself to avoid needing all of these exclusions.

Reply:
Yes, I enabled that checker a couple of weeks ago and I am still not sure how to configure it in the right way for a Jenkins plugin. It works quite well with enabled dependencies for my analysis-model package, however here I'm still experimenting. I think disabling dependencies is a good alternative to get the focus on the changes in the plugin.